### DISCUSSION TITLE
KV igbinary

### DISCUSSION THREAD

**Discussion by konigbach** [spiral/spiral#1055] [Q&A] [answered]
Hello!
I've read that rr can manage igbinary serializer out of the box. 
Is there a configuration to make this in spiral?
```
(new Factory($rpc)
    ->withSerializer(new IgbinarySerializer())
    ->select('storage');
```
Thanks!

**Comment by butschster** [comment, marked as answer]
Hil @konigbach 

To use `IgbinarySerializer` in spiral framework you just need to [rebind](https://spiral.dev/docs/framework-container/current/en#configuring-container) a [default serializer](https://github.com/spiral/roadrunner-bridge/blob/3.x/src/Bootloader/CacheBootloader.php#L33) for KV plugin.

Something like

```php
<?php

declare(strict_types=1);

namespace App\Application\Bootloader;

use Spiral\Boot\Bootloader\Bootloader;
use Spiral\RoadRunner\KeyValue\Serializer\IgbinarySerializer;
use Spiral\RoadRunner\KeyValue\Serializer\SerializerInterface;

final class AppBootloader extends Bootloader
{
    public function defineSingletons(): array
    {
        return [
            SerializerInterface::class => IgbinarySerializer::class,
        ];
    }
}
```

**Reply by konigbach** [reply]
Thank you for your quick response!
I get it but how should I use the Cache? I understand that there's a PSR CacheInterface which doesn't provide this functionality. Should I use "Spiral\Cache\CacheStorageProviderInterface" instead? I'm a little bit confused about Cache/Storage being together.

This was the first approach:

```
final class RedisProductPriceRepository
{
    public function __construct(
        private readonly Psr\SimpleCache\CacheInterface\CacheInterface $cache,
    ) {
    }

    public function save(Product $product): void
    {
        $this->cache->set(
            sprintf(
                '%s_%s',
                CacheKeys::Products->value,
                $product->id(),
            ),
            igbinary_serialize($product->values()),
        );
    }
}
```
But now I'd like to force always to use igbinary. I appreciate any help.Thank you again!

**Reply by butschster** [reply]
All you need is just 


```php
$this->cache->set(
    key: 'key', 
    value: ['some' => 'data'], 
    ttl: 3600
);
```

Values will be serialized authomaticaly

https://spiral.dev/docs/basics-cache/current/en#usage

**Reply by konigbach** [reply]
Thank you, man!!
